tests/permfun.R

```diff
@@ -0,0 +1,14 @@
+## Source: https://stackoverflow.com/questions/11095992/generating-all-distinct-permutations-of-a-list-in-r
+permutations <- function(n){
+  if(n==1){
+    return(matrix(1))
+  } else {
+    sp <- permutations(n-1)
+    p <- nrow(sp)
+    A <- matrix(nrow=n*p,ncol=n)
+    for(i in 1:n){
+      A[(i-1)*p+1:p,] <- cbind(i,sp+(sp>=i))
+    }
+    return(A)
+  }
+}
```

Comment:
I wasn't sure, because it's only used by the tests and not the functions in the package itself. But agree that it makes more sense, so will move it there.

Reply:
I thought the same thing for a second as well, but eventually decided that misc.R is a good place to put all internal functions, no matter whether they are used by external functions or just the unit tests.